dom.go

```diff
@@ -911,14 +910,17 @@ func doCopy(c Component) Component {
 }
 
 // copyProps copies all struct fields from src to dst that are tagged with
-// `vecty:"prop"`.
+// `vecty:"prop"`, then replaces src with dst.
 //
-// If src and dst are different types, copyProps is no-op.
+// If src and dst are different types or non-pointers, copyProps panics.
 func copyProps(src, dst Component) {
 	s := reflect.ValueOf(src)
 	d := reflect.ValueOf(dst)
 	if s.Type() != d.Type() {
-		return
+		panic("vecty: internal error (attempted to copy properties of incompatible structs)")
+	}
+	if s.Kind() != reflect.Ptr || d.Kind() != reflect.Ptr {
```

Comment:
I suspect that if the component was not a pointer, subsequent `.Elem()` calls would have failed with a panic anyway, and we need this here to ensure that we can perform the `Set()` operation later.

Reply:
Makes sense.